### ISSUE TITLE
Container publish not functioning in DN8.P2; Worked in DN7

### ISSUE THREAD

**Issue by MarkStega** [dotnet/aspnetcore#47259]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

I am attempting to publish to a container using a workflow running in github actions. This publish worked in DN7, but fails in DN8.P2 with an error logged of 'The "CreateNewImage" task failed unexpectedly.'

### Expected Behavior

I expect the publish to complete successfully

### Steps To Reproduce

The steps in the workflow that are pertinent are:
```
    - name: Docker login
      uses: azure/docker-login@v1
      with:
        login-server: ${{ secrets.REGISTRY_LOGIN_SERVER }}
        username: ${{ secrets.REGISTRY_USERNAME }}
        password: ${{ secrets.REGISTRY_PASSWORD }}

    - name: Publish, tag, and push DS image
      run: |
        dotnet publish ICEBG.Web.DataServices/ICEBG.Web.DataServices.csproj --os linux --arch x64 --configuration Azure -p:PublishProfile=DefaultContainer -p:Version=${{ env.suffix }} -p:ContainerImageName=${{env.appNameDS}}
        docker tag ${{env.appNameDS}}:${{ env.suffix }} ${{ secrets.REGISTRY_LOGIN_SERVER }}/${{env.appNameDS}}:${{ env.suffix }}
        docker push ${{ secrets.REGISTRY_LOGIN_SERVER }}/${{env.appNameDS}}:${{ env.suffix }}
```

The failure occurs in the 'dotnet publish'  step

The logs show
```
  ICEBG.Client -> /home/runner/work/ICEBG/ICEBG/ICEBG.Client/bin/Azure/net8.0/ICEBG.Client.dll
  ICEBG.Web.DataServices -> /home/runner/work/ICEBG/ICEBG/ICEBG.Web.DataServices/bin/Azure/net8.0/linux-x64/ICEBG.Web.DataServices.dll
  ICEBG.Web.DataServices -> /home/runner/work/ICEBG/ICEBG/ICEBG.Web.DataServices/bin/Azure/net8.0/linux-x64/publish/
/home/runner/.nuget/packages/microsoft.net.build.containers/0.3.2/build/Microsoft.NET.Build.Containers.targets(114,9): error MSB4018: The "CreateNewImage" task failed unexpectedly. [/home/runner/work/ICEBG/ICEBG/ICEBG.Web.DataServices/ICEBG.Web.DataServices.csproj]
/home/runner/.nuget/packages/microsoft.net.build.containers/0.3.2/build/Microsoft.NET.Build.Containers.targets(114,9): error MSB4018: System.AggregateException: One or more errors occurred. (Response status code does not indicate success: 404 (Not Found).) [/home/runner/work/ICEBG/ICEBG/ICEBG.Web.DataServices/ICEBG.Web.DataServices.csproj]
```
The repository is open source and available at [https://github.com/MarkStega/ICEBG](https://github.com/MarkStega/ICEBG)

The full log of the failed run is available at [https://github.com/MarkStega/ICEBG/actions/runs/4439861459/jobs/7792909114](https://github.com/MarkStega/ICEBG/actions/runs/4439861459/jobs/7792909114)

### .NET Version

DN8.P2

**Comment by adityamandaleeka** [comment]
@MichaelSimons @richlander Can you help route this?

**Comment by richlander** [comment]
This looks like @baronfel

**Comment by baronfel** [comment]
Yep, we've got an AzDo ticket for this that we recently fixed in https://github.com/dotnet/sdk-container-builds/pull/384. We're going to publish another stable version of the package to NuGet likely this week that will contain this fix.

@MarkStega In the meantime you can fix this in one of two ways:
* Explicitly specify the `ContainerBaseImage` yourself: `<ContainerBaseImage>mcr.microsoft.com/dotnet/aspnet:8.0-preview</ContainerBaseImage>`
* Use the nightly builds in the repository [as described in this document](https://github.com/dotnet/sdk-container-builds/blob/main/DEVGUIDE.md#using-the-prerelease-package-from-this-repository) to use a from-main-branch build that contains the fix.

Apologies for the inconvenience, we were behind in reacting to the new 8.0 tagging scheme.

**Comment by MarkStega** [comment]
@baronfel Thanks, I added the container base image and all is well. Now I just have to remember to take those out of the two csproj's when you release a new container package!

**Comment by baronfel** [comment]
@MarkStega from that log it looks like the port used in your app is 80 - in .NET 8 the containers' default ports changed to 8080 (per [this change](https://github.com/dotnet/dotnet-docker/issues/3968)) and I think that's what's causing your startup issue. Try updating your service's config to 8080, or overriding the urls in use. This can be done by explicitly setting the ASPNETCORE_URLS environment variable, or by setting the new ASPNETCORE_HTTP_PORTS or ASPNETCORE_HTTPS_PORTS variables (these latter two only get used when the ASPNETCORE_URLS variable isn't used). You can set container environment variables using the SDK tooling - check [the docs](https://github.com/dotnet/sdk-container-builds/blob/main/docs/ContainerCustomization.md#containerenvironmentvariable) for examples on that.

**Comment by MarkStega** [comment]
@baronfel 

Chet,
It was timely that you published your .Net 7 SDK container article. I've updated my project files to use the new tooling. The link that you had to the rootless containers contained exactly what I needed to do in Azure, namely, setting WEBSITES_PORT to 8080 (and then waiting quite a while for it to propagate so that the docker run used that port rather than 80).

I had to laugh though when I looked at your repository's (sdk-container-demo) actions log. I thought it was only me that took many, many revisions to get to a working :-)

Thanks for the help and keep going on smoothing the direct publish of images from the sdk.
